pkg/config/tlscfg/testdata/gen-certs.sh

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env bash
+
+# The following commands were used to create the CA, server and client's certificates and keys in this directory used by unit tests.
+# These certificates use the Subject Alternative Name extension rather than the Common Name, which will be unsupported in Go 1.15.
+
+usage() {
+  echo "Usage: $0 [-d]"
+  echo
+  echo "-d  Dry-run mode. PEM files will not be modified."
+  exit 1
+}
+
+dry_run=false
+
+while getopts "d" o; do
+    case "${o}" in
+        d)
+            dry_run=true
+            ;;
+        *)
+            usage
+            ;;
+    esac
+done
+shift $((OPTIND-1))
+
+set -ex
+
+# Create temp dir for generated files.
+tmp_dir=$(mktemp -d -t certificates)
+clean_up() {
+    ARG=$?
+    if [ $dry_run = true ]; then
+      echo "Dry-run complete. Generated files can be found in $tmp_dir"
+    else
+      rm -rf "$tmp_dir"
+    fi
+    exit $ARG
+}
+trap clean_up EXIT
+
+# Generate config files.
+# The server name (under alt_names in the ssl.conf) is `example.com`. (in accordance to [RFC 2006](https://tools.ietf.org/html/rfc2606))
+source gen-ssl-conf.sh example.com "$tmp_dir/ssl.conf"
+source gen-ssl-conf.sh wrong.com "$tmp_dir/wrong-ssl.conf"
```

Comment:
Why do you need to use `source` instead of just invoking the script?

Also, since it just generates a config file, can you not define it as a function in this single script file?

Reply:
`source` will allow `set -ex` to take affect in the `source`-ed script.

The only reason for keeping it as a separate script was to separate the concern of config file generation from certificate generation, which may be useful for debugging. However, the file generating code is not particularly complex, so it can be moved into a function of the main script.